core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAOSpec.scala

```diff
@@ -122,4 +128,83 @@ class HttpGoogleServicesDAOSpec extends AnyFlatSpec with Matchers with MockitoTe
     await(httpGoogleServicesDao.listBillingAccounts(userInfo, Some(true))) shouldBe List(billingAccountWithAccess)
     await(httpGoogleServicesDao.listBillingAccounts(userInfo, Some(false))) shouldBe List(billingAccountNoAccess)
   }
+
+  behavior of "setupWorkspace"
+
+  it should "create the workspace's GCS bucket" in {
+    val userInfo = UserInfo(RawlsUserEmail("fake@email.com"),
+                            OAuth2BearerToken("some-token"),
+                            300,
+                            RawlsUserSubjectId("193481341723041"),
+                            None
+    )
+
+    val googleStorageService = mock[GoogleStorageService[IO]](RETURNS_SMART_NULLS)
+    val googleProjectId = "project-id"
+    val bucketName = GcsBucketName("fc-bucket-name")
+    when(
+      googleStorageService.insertBucket(
+        ArgumentMatchers.eq(GoogleProject(googleProjectId)),
+        ArgumentMatchers.eq(bucketName),
+        any(),
+        any(),
+        any(),
+        any(),
+        any(),
+        any(),
+        any(),
+        any(),
+        autoclassEnabled = ArgumentMatchers.eq(true)
+      )
+    ).thenReturn(fs2.Stream.unit)
+
+
+    val googleServicesDAO = new HttpGoogleServicesDAO(
```

Comment:
🤯 at the parameters list for this DAO. 🌱 for the future: we should refactor to a sane set of collaborating objects (i.e., an HttpGoogleServicesBillingDAO, HttpGoogleServicesStorage DAO ,etc.)